app/scripts/modules/core/src/application/service/inferredApplicationWarning.service.ts

```diff
@@ -0,0 +1,48 @@
+import { module } from 'angular';
+
+import { Application } from 'core/application/application.model';
+import { NOTIFIER_SERVICE, NotifierService } from 'core/widgets/notifier/notifier.service';
+
+export class InferredApplicationWarningService {
+
+  private viewedApplications: string[] = [];
+
+  constructor(private notifierService: NotifierService) {
+    'ngInject';
+  }
+
+  public resetViewedApplications(): void {
+    this.viewedApplications.length = 0;
+  }
+
+  public isInferredApplication(app: Application): boolean {
+    return !app.attributes || !app.attributes.email;
+  }
+
+  public checkIfInferredAndWarn(app: Application): void {
+    if (this.check(app)) {
+      this.warn(app.name);
+    }
+  }
+
+  private check(app: Application): boolean {
+    const { name } = app;
+    const hasViewed = this.viewedApplications.includes(name);
+
+    this.viewedApplications.push(name);
```

Comment:
Is there a concern that this could grow unbounded?

Reply:
Not really. If a user looks at a thousand applications in the same session, they'll end up with a thousand strings in the array, which isn't going to cause any performance problems.